internal/datastore/postgres/migrations/driver.go

```diff
@@ -26,7 +26,7 @@ type AlembicPostgresDriver struct {
 }
 
 // NewAlembicPostgresDriver creates a new driver with active connections to the database specified.
-func NewAlembicPostgresDriver(ctx context.Context, url string) (*AlembicPostgresDriver, error) {
+func NewAlembicPostgresDriver(ctx context.Context, url string, awsIamAuthEnabled bool, awsSdkConfig aws.Config) (*AlembicPostgresDriver, error) {
```

Comment:
Rather than taking in AWS configuration here, which would tie the implementation to AWS, can we accept an optional "CredentialsProvider" or "TokenProvider" interface implementation, and then have the command entrypoint construct an AWS-specific version of that interface? I could also see it being very useful for the MySQL driver as well